assets/js/base/components/quantity-selector/index.js

```diff
@@ -13,30 +19,121 @@ const QuantitySelector = ( {
 	className,
 	quantity = 1,
 	onChange = () => null,
+	itemName = '',
 } ) => {
-	const classes = classNames(
-		'wc-block-quantity-selector__input',
-		className
+	const classes = classNames( 'wc-block-quantity-selector', className );
+	const inputRef = useRef( null );
+	const [ currentValue, setCurrentValue ] = useState( quantity );
+
+	useEffect( () => {
+		if ( currentValue !== inputRef.current.value ) {
+			inputRef.current.value = currentValue;
+		}
+		onChange( currentValue );
+	}, [ currentValue, onChange ] );
+
+	/**
+	 * Handles keyboard up and down keys to change quantity value.
+	 *
+	 * @param {Object} event event data.
+	 */
+	const quantityInputOnKeyDown = useCallback(
+		( event ) => {
+			if ( event.keyCode === KEYCODE_UP ) {
+				event.preventDefault();
+				setCurrentValue( currentValue + 1 );
+			}
+
+			if ( event.keyCode === KEYCODE_DOWN && currentValue > 0 ) {
+				event.preventDefault();
+				setCurrentValue( currentValue - 1 );
+			}
+		},
+		[ currentValue, setCurrentValue ]
 	);
 
-	// For now just use a regular number edit. (temporary)
-	// @todo https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1521
 	return (
-		<input
-			className={ classes }
-			type="number"
-			step="1"
-			min="0"
-			value={ quantity }
-			onChange={ ( event ) => onChange( event.target.value ) }
-		/>
+		<div className={ classes }>
+			<input
+				className="wc-block-quantity-selector__input"
+				type="number"
+				step="1"
+				min="0"
+				value={ quantity }
+				ref={ inputRef }
+				onKeyDown={ quantityInputOnKeyDown }
+				onChange={ ( event ) => {
+					const value =
+						isNaN( event.target.value ) || ! event.target.value
```

Comment:
Just curious, is the `! event.target.value` check necessary?

Reply:
Yes because `isNaN( '' )` returns false. `parseInt` throws errors with empty strings.

Reply:
TIL, thanks!